File: Python/fundamentals/bubble_sort/bubble_sort.py
# bubble sort 
p = lambda x : print(x)
arr = [x for x in range(15,0,-1)]

def bubble_sort(arr):
    if len(arr) <= 1:
        return arr 
    _bool = True 
    while _bool : 
        _bool = False 
        for idx in range(len(arr)-1):
            if arr[idx] > arr[idx+1]:
                temp = arr[idx]
                arr[idx] = arr[idx+1]
                arr[idx+1] = temp 
                _bool = True
    return arr 

p(bubble_sort(arr)) 
             

File: MEAN/PreMean/js_fundamentals/bubbleSort.js
// bubble sort 




var bubbleSort = function(arr){
    var runner = true 
    while(runner){
        runner = false 
        for(var i = 0; i < arr.length-1; i++){
            if(arr[i]>arr[i+1]){
                runner = true 
                var temp = arr[i]
                arr[i] = arr[i+1]
                arr[i+1] = temp 
            }
        }
    }
    return arr 
}

var arr = [...Array(40)].map(x => Math.round(Math.random()*1000))
var sorted = bubbleSort(arr)
console.log(sorted)